app/models/miq_ae_datastore.rb

```diff
@@ -170,17 +170,15 @@ def self.reset_to_defaults
   end
 
   def self.reset_domains_from_legacy_directory
-    domain_files = DATASTORE_DIRECTORY.join('*', MiqAeDomain::DOMAIN_YAML_FILENAME)
-    Dir.glob(domain_files).each do |domain_file|
-      domain_name = File.basename(File.dirname(domain_file))
-      reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant)
-    end
+    legacy_domains.each { |domain_name| reset_domain(DATASTORE_DIRECTORY, domain_name, Tenant.root_tenant) }
   end
 
   def self.reset_domains_from_vmdb_plugins
-    datastores_for_reset.each do |domain|
-      reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant)
-    end
+    plugin_domain_names.each { |domain| reset_domain(domain.datastores_path.to_s, domain.name, Tenant.root_tenant) }
```

Comment:
- `plugin_domain_names` should really be renamed to `plugin_domains` since it returns objects.
- `legacy_domains` also returns object, and so could use `domain.datastores_path.to_s` instead of `DATASTORE_DIRECTORY`

So given both of those, these 2 methods become identical, and we can refactor them away.